src/material/core/common-behaviors/common-module.ts

```diff
@@ -90,22 +91,36 @@ export class MatCommonModule {
     }
   }
 
+    /** Access injected document if available or fallback to global document reference */
+    private _getDocument(): Document | null {
+      const doc = this._document || document;
+      return typeof doc === 'object' && doc ? doc : null;
```

Comment:
this check isn't in the other `_getDocument` methods in this PR

Reply:
True, the check here was already present so I was just preserving to maintain any logic based off it.

I'm happy to add this same check everywhere else though if we want.

Reply:
Nah seems like overkill, lets just leave it for this one because it had it. Eventually we can just remove all this logic and rely on the injected value anyways...